src/components/TodoActions/index.tsx

```diff
@@ -1,44 +1,94 @@
 import { Box, Button, CardActions, Typography, TextField } from "@mui/material";
 import { useForm, SubmitHandler, FieldValues } from "react-hook-form";
 import { yupResolver } from "@hookform/resolvers/yup";
-import { schema } from "../../utils/data";
-import todoActionsStyle from "./todoActions";
+import { useEffect } from "react";
+
+import { todoSchema } from "../../utils/data";
+import todoActionsStyle from "./styles";
+import { Todo, addTodo, setTodoItem, updateTodo } from "../../store/todoSlice";
+import { useSelector, useDispatch } from "react-redux";
+import { RootState } from "../../store/store";
 
 function TodoActions() {
+  const selectedTodo = useSelector(
+    (state: RootState) => state.todoList.todoItem
+  );
+  const dispatch = useDispatch();
```

Comment:
instead of useSelector and useDispatch use redux connect through container pattern.